cpp/src/Slice/Scanner.l

```diff
@@ -92,6 +92,7 @@ namespace
 
   /* List of start-condition states the scanner can be in. This lets the scanning be context dependent. */
```

Comment:
At any moment the Slice scanner is in a specific state (called a "Start Condition").
This list contains all the possible states that it can be within.

Previously, we only had a single C_COMMENT state, and whenever the scanner was in this state, it would add any text it saw to the current doc-comment. This was bad.
So, now we have 2 states. C_DOC_COMMENT is triggered when the scanner sees /** and C_COMMENT is triggered when the scanner sees /*. The scanner will try to match rules in order so it will always try to match /** first.

And, the scanner will only add things to doc comments while it is in the C_DOC_COMMENT state.